packages/plugin-ext/src/main/browser/view/tree-views-main.ts

```diff
@@ -131,3 +139,21 @@ export class TreeViewsMainImpl implements TreeViewsMain, Disposable {
     }
 
 }
+
+@injectable()
+export class TreeViewsMainServiceProvider implements RPCProtocolServiceProvider {
```

Comment:
I wonder can `TreeViewsMainImpl` implement `RPCProtocolService` instead? `RPCProtocolService` should have a mandatory property like `rpcIdentifier`. Then we can get rid of all `RPCProtocolServiceProvider`. It is much less code.